R/types.R

```diff
@@ -432,11 +431,57 @@ coerce_class <- function(.x, .to, ...) {
   switch(class(.x), ..., abort_coercion(.x, .to))
 }
 abort_coercion <- function(x, to) {
-  abort(paste0(
-    "Cannot convert objects of type `", type_of(x), "` to `", to, "`"
-  ))
+  x_type <- friendly_type(type_of(x))
+  to_type <- friendly_type(to)
+  abort(paste0("Cannot convert ", x_type, " to ", to_type))
 }
 
+#' Format a type for error messages
+#'
+#' @param type A type as returned by [type_of()].
+#' @param article A string, either `"indefinite"` or
+#'   `"definite"`. Indefinite articles depend on the starting letter
+#'   of `type`: `"an"` if starts with a vowel, `"a"` otherwise.
+#' @export
+#' @examples
+#' friendly_type("logical")
+#' friendly_type("integer")
+#' friendly_type("string")
+friendly_type <- function(type, article = c("indefinite", "definite")) {
+  article <- match.arg(article)
+
+  type <- switch(type,
+    logical = ,
+    integer = ,
+    double = ,
+    complex = ,
+    character = ,
+    raw =
+      paste(type, "vector"),
+    language = "language call",
```

Comment:
I think as a general principle it's better to right code like:

```R
switch(type, 
  logical = "a logical vector",
  integer = "an integer vector", 
  ...
)
```

Otherwise it's a basically impossible to translate to other languages.